### PR TITLE
MAINT: spatial: distance refactoring

### PR DESCRIPTION
**Pull request by larsmans**

### CODE REVIEW

File: scipy/spatial/src/distance_impl.h [hunk 1]
```diff
@@ -344,13 +339,13 @@ static NPY_INLINE double
...
-  for (i = 0; i < n; i++) {
-    d = fabs(u[i] - v[i]) * w[i];
-    s = s + pow(d, p);
-  }
-  return pow(s, 1.0 / p);
+    npy_intp i = 0;
+    double s = 0.0, d;
+    for (i = 0; i < n; i++) {
+        d = fabs(u[i] - v[i]) * w[i];
+        s = s + pow(d, p);
```
Comment (perimosocordiae) [inline scipy/spatial/src/distance_impl.h]: Why not `s += pow(d, p)`?

Comment (larsmans) [inline scipy/spatial/src/distance_impl.h]: Done now.

**Comment by ev-br** [comment]
From the peanut gallery: is gh-4668 anyhow relevant/related/wanted here?

**Comment by larsmans** [comment]
@ev-br Not that I know of, are you sure that's the right issue number?

**Comment by pv** [comment]
thanks, lgtm. there was some weird stuff there...

**Pull request merged**